# Ejercicio 2 analisis de datos

In [41]:
# Importo librerias
import numpy as np
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

In [61]:
import pandas as pd

# URL del archivo en tu repositorio de GitHub
url = "https://raw.githubusercontent.com/Jevillalbav/parcial_final_poo/main/airlines_reviews.csv"

# Leer el archivo directamente usando pandas
df = pd.read_csv(url)
df.columns = [i.replace(' ', '_').lower() for i in df.columns]

Se sugiere que antes de comenzar con el análisis se lea cada una de las descripciones de
las columnas para tener un contexto general de los datos.

Primero, se requiere poder dar respuesta a lo siguiente:

    a. ¿Cuántos comentarios se encuentran verificados?
    b. ¿Cuál es el rango de fechas de los vuelos que se registran en el dataset?
    c. ¿Cuáles aerolíneas registran por lo menos un comentario?
    d. ¿Cuál es el promedio de comentarios de cada aerolínea por año?


In [60]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8100 entries, 0 to 8099
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   title                   8100 non-null   object
 1   name                    8100 non-null   object
 2   review_date             8100 non-null   object
 3   airline                 8100 non-null   object
 4   verified                8100 non-null   int64 
 5   reviews                 8100 non-null   object
 6   type_of_traveller       8100 non-null   object
 7   month_flown             8100 non-null   object
 8   route                   8100 non-null   object
 9   class                   8100 non-null   object
 10  seat_comfort            8100 non-null   int64 
 11  staff_service           8100 non-null   int64 
 12  food_&_beverages        8100 non-null   int64 
 13  inflight_entertainment  8100 non-null   int64 
 14  value_for_money         8100 non-null   int64 
 15  over

In [62]:
df.isna().sum()

title                     0
name                      0
review_date               0
airline                   0
verified                  0
reviews                   0
type_of_traveller         0
month_flown               0
route                     0
class                     0
seat_comfort              0
staff_service             0
food_&_beverages          0
inflight_entertainment    0
value_for_money           0
overall_rating            0
recommended               0
dtype: int64

In [63]:
df.sample(7).T

6676  \
title                                      avoid Air France in the future   
name                                                             J Darson   
review_date                                                    2018-02-16   
airline                                                        Air France   
verified                                                             True   
reviews                   Moscow to London via Paris. I was less than ...   
type_of_traveller                                            Solo Leisure   
month_flown                                                 February 2018   
route                                      Lisbon to Washington via Paris   
class                                                       Economy Class   
seat_comfort                                                            3   
staff_service                                                           2   
food_&_beverages                                                        5   
inflight_entertainment                                                  4   
value_for_money                                                         4   
overall_rating                                                          3   
recommended                                                            no   

                                                                     1901  \
title                                             update the wrong Qmiles   
name                                                         Ayub Mohamed   
review_date                                                    2019-05-02   
airline                                                     Qatar Airways   
verified                                                             True   
reviews                   Birmingham to Doha. Qatar Airways is the bes...   
type_of_traveller                                          Family Leisure   
month_flown                                                    April 2019   
route                                     Stockholm to Singapore via Doha   
class                                                       Economy Class   
seat_comfort                                                            2   
staff_service                                                           5   
food_&_beverages                                                        3   
inflight_entertainment                                                  3   
value_for_money                                                         3   
overall_rating                                                          7   
recommended                                                           yes   

                                                                     3596  \
title                                            they ignore your request   
name                                                          Kenghua Tai   
review_date                                                    2018-08-23   
airline                                                          Emirates   
verified                                                             True   
reviews                   Bahrain to Singapore via Dubai. Almost misse...   
type_of_traveller                                            Solo Leisure   
month_flown                                                   August 2018   
route                                         Amsterdam to Male via Dubai   
class                                                       Economy Class   
seat_comfort                                                            5   
staff_service                                                           5   
food_&_beverages                                                        4   
inflight_entertainment                                                  5   
value_for_money                                                         2   
overall_rating                                                 

-----------

### ¿ Cuantos comentarios se encuentran verificados ? 

In [64]:
## se cuentan valores
df['verified'].value_counts()

True                                                                                                                                                                                                                                           6216
False                                                                                                                                                                                                                                          1881
*Unverified*                                                                                                                                                                                                                                      1
NotVerified                                                                                                                                                                                                                                       1
we do appreciate you bri

In [65]:
# se revisa que paso con los que no son True o False
df.query('verified != "True" & verified != "False"')

title          name review_date  \
2225     CSD was professional but friendly        D Webb  2017-11-18   
3358                     Very disappointed        HY Teo  2019-08-13   
3822  they allocated us to different seats  John Perrins  2017-11-19   

            airline                                           verified  \
2225  Qatar Airways                                       *Unverified*   
3358       Emirates                                        NotVerified   
3822       Emirates  we do appreciate you bringing this matter to o...   

                                                reviews type_of_traveller  \
2225    Delhi to London via Doha. The return flight ...      Solo Leisure   
3358    Singapore to Barcelona via Dubai with Emirat...    Couple Leisure   
3822    Birmingham to Johannesburg via Dubai. We pai...    Couple Leisure   

         month_flown                             route          class  \
2225   November 2017                      Kiev to Doha    First Class   
3358     August 2019  Melbourne to Frankfurt via Dubai  Economy Class   
3822  September 2017             Narita Tokyo to Dubai  Economy Class   

      seat_comfort  staff_service  food_&_beverages  inflight_entertainment  \
2225             5              4                 5                       4   
3358             3              3                 1                       2   
3822             2              5                 4                       4   

      value_for_money  overall_rating recommended  
2225                3               7         yes  
3358                3               5          no  
3822                1               2          no

In [69]:
conditions = [
    df['verified'] == 'True',
    df['verified'] == 'False',
    df['verified'] == 'NotVerified',
    df['verified'] == '*Unverified*']
    
df['verified_'] = np.select(conditions, ['True', 'False', 'False', 'False'], default= np.nan)

In [70]:
df['verified_'].value_counts().to_frame()

verified_
True        6216
False       1883
nan            1

In [71]:
df['verified_'].value_counts(normalize=True).to_frame().round(2)

verified_
True        0.77
False       0.23
nan         0.00

## Respuesta:  
 
Los 3 casos diferentes de True o False en la columna verified fueron revisados, dos de ellos son unverified y not verified por lo cual es facil asumir que estos no han sido verificados, el tercer caso es distinto por lo cual no se asume nada y se da como mismatched.  

verified_

**True**	6216 casos

**False**	1883 casos 

**nan**	1 casos

-----------

## Cual es el rango de fechas de los vuelos que se registran en el dataset?

In [86]:
dates = df['month_flown'].str.split(' ', expand=True).copy()
dates.columns = ['month', 'year']
dates['month'] = dates['month'].map({'January': 1, 'February': 2, 'March': 3, 'April': 4, 'May': 5, 'June': 6, 'July': 7, 'August': 8, 'September': 9, 'October': 10, 'November': 11, 'December': 12})
dates['year'] = dates['year'].astype(int)
dates['date'] = pd.to_datetime(dates.assign(day=1)[['year', 'month', 'day']])

first = dates['date'].min()
last = dates['date'].max()

print(f'First date: {first}')
print(f'Last date: {last}')

First date: 2013-03-01 00:00:00
Last date: 2024-03-01 00:00:00


## Respuesta:  

El rango de fechas esta entre Marzo de 2013 y Marzo de 2024